src/main/daml/Daml/Finance/Settlement/Batch.daml

```diff
@@ -82,7 +83,13 @@ template Batch
         forA_ (sortAndGroupOn fst $ catOptionals =<< usedInterfaceTypeReps)
           \ts -> assertMsg "Same instrument's holdings must have same template type."
             $ S.size (S.fromList ts) == 1
-        pure $ catOptionals settledCids
+        -- sort retun settledCids according to the initial order of the instructions
+        let
+          keys = instructionIds
+          map = M.fromList $ zip (fmap (.id) orderedInstructions) settledCids
```

Comment:
The `map` defined here could be confused with the function `map` from the stdlib. 

```suggestion
          cidMap = M.fromList $ zip (fmap (.id) orderedInstructions) settledCids
```

Reply:
renamed it